diff --git a/src/aioquic/quic/packet.py b/src/aioquic/quic/packet.py
index ce2b6c5..b4e7225 100644
--- a/src/aioquic/quic/packet.py
+++ b/src/aioquic/quic/packet.py
@@ -628,13 +628,6 @@ def push_ack_frame(buf: Buffer, rangeset: RangeSet, delay: int) -> int:
     r = rangeset[index]
     buf.push_uint_var(r.stop - 1)
     buf.push_uint_var(delay)
-    buf.push_uint_var(index)
-    buf.push_uint_var(r.stop - 1 - r.start)
-    start = r.start
-    while index > 0:
-        index -= 1
-        r = rangeset[index]
-        buf.push_uint_var(start - r.stop - 1)
-        buf.push_uint_var(r.stop - r.start - 1)
-        start = r.start
+    buf.push_uint_var(0)
+    buf.push_uint_var(r.stop - 1 - rangeset[0].start)
     return ranges
